### PR TITLE
First go at install text

### PR DESCRIPTION
**Pull request by w3168**
Almost  ready to go. Based off Thetis and Icepack installation pages. 

Should we add something about docker, similar to the icepack website?

When there is a link for the first tutorial we can add a hyperlink there.

### CODE REVIEW

**Review by angus-g — CHANGES_REQUESTED** [review]
Nice! Given that this will link to a tutorial/demo, I guess we'll hold off on the merge until those are coming in properly. Other than that, a couple of minor points.

File: docs/install.md [hunk 1]
```diff
@@ -1,5 +1,44 @@
...
+First of all, you need to install [Firedrake](https://www.firedrakeproject.org), the finite element framework that G-ADOPT is built on.
+
+Firedrake is available for Ubuntu, Mac and in principle other Linux and Linux-like systems. 
+You can install Firedrake by running: 
+
+    curl -O https://raw.githubusercontent.com/firedrakeproject/firedrake/master/scripts/firedrake-install
+    python3 firedrake-install
+
+More detailed instructions for installing Firedrake and suggestions for troubleshooting can be found 
+on the [Firedrake website](https://www.firedrakeproject.org/download.html). Note that the install proceduce may take
```
[the remark this answers is not in the document]
Comment (w3168) [reply docs/install.md]: thanks I've fixed this

File: docs/install.md [hunk 2]
```diff
@@ -1,5 +1,44 @@
...
+On Ubuntu and similar systems, you can obtain Paraview by installing the paraview package. On Mac OS the 
+easiest approach is to download a binary from the paraview website.
+
+## Optional Install
+
+To bring in the optional nonlinear optimisation dependencies, install the optimisation variant:
+
+    pip install gadopt[optimisation]
+
+For developers who intend to change the core G-ADOPT code you should add the editable flag:
```
[the remark this answers is not in the document]
Comment (w3168) [reply docs/install.md]: Thanks I've fixed this

**Comment by w3168** [comment]
Thanks @drhodrid I've updated the markdown so that the default installation is using the --install method. I've also removed  the (non-existent) hyperlink to the tutorials section to be ready later

@angus-g I've kept the section about specifying the optimisation argument because Firedrake won't install this by default. (and said that you need have the venv activated with an existing firedrake installation) - does that sound right?

I've checked the above install works ok (it was actually very quick). I ran the base case and everything seemed fine!

Firedrake does automatically use the -e option for pip install based on the directory in /firedrake/src/g-adopt. (I checked by modifying stokes integrators slightly). @angus-g do you think we should keep the separate 'developing g-adopt' section? 

As @angus-g said in the meeting the latest pip install overrides previous gadopt packages so it shouldnt break anything if someone trys to install it twice...

I also thought that it would be best not to add the firedrake-update --install gadopt option (e.g. see the thetis install page), because in my experience as a user when I haven't updated or installed a new version of Firedrake recently using firedrake-update often breaks... So in my opinion it would probably be better to just advertise the install option...

Also as discussed in the meeting we will add the gadi/setonix instructions for running (but not build recipes for the sake of Angus' sanity!!) in a subsequent pull request

**Comment by angus-g** [comment]
> I've kept the section about specifying the optimisation argument because Firedrake won't install this by default. (and said that you need have the venv activated with an existing firedrake installation) - does that sound right?

Yep, that's right.

> do you think we should keep the separate 'developing g-adopt' section?

I think it is probably better to be explicit about this, and probably better/easier to be developing outside of the Firedrake venv, so I'd say keep it.

> I also thought that it would be best not to add the firedrake-update --install gadopt option (e.g. see the thetis install page), because in my experience as a user when I haven't updated or installed a new version of Firedrake recently using firedrake-update often breaks... So in my opinion it would probably be better to just advertise the install option...

Agreed, better to just pip install it in the venv for an existing Firedrake install

**Review by angus-g — APPROVED** [review]
I think we're good, thanks!

**Pull request merged**